core/src/test/scala/integration/kafka/api/SaslSetup.scala

```diff
@@ -100,6 +100,24 @@ trait SaslSetup {
     }
   }
 
+  protected def jaasSectionsInvalidClientCredentials(kafkaServerSaslMechanisms: Seq[String],
+                                                     kafkaClientSaslMechanism: Option[String],
+                                                     mode: SaslSetupMode = Both,
+                                                     kafkaServerEntryName: String = JaasTestUtils.KafkaServerContextName): Seq[JaasSection] = {
+    val hasKerberos = mode != ZkSasl &&
+      (kafkaServerSaslMechanisms.contains("GSSAPI") || kafkaClientSaslMechanism.exists(_ == "GSSAPI"))
+    if (hasKerberos)
+      maybeCreateEmptyKeytabFiles()
+    mode match {
+      case ZkSasl => JaasTestUtils.zkSectionsInvalidCredentials
+      case KafkaSasl =>
+        Seq(JaasTestUtils.kafkaServerSection(kafkaServerEntryName, kafkaServerSaslMechanisms, serverKeytabFile),
+          JaasTestUtils.kafkaClientSectionInvalidCredentials(kafkaClientSaslMechanism, clientKeytabFile))
+      case Both => Seq(JaasTestUtils.kafkaServerSection(kafkaServerEntryName, kafkaServerSaslMechanisms, serverKeytabFile),
+        JaasTestUtils.kafkaClientSectionInvalidCredentials(kafkaClientSaslMechanism, clientKeytabFile)) ++ JaasTestUtils.zkSections
+    }
+  }
```

Comment:
This method is not required. For invalid client credentials, you can simply override `sasl.jaas.config` property